### ISSUE TITLE
Psalm should not report MissingFile if file is checked if it exists before include/require

### ISSUE THREAD

**Issue by filips123** [vimeo/psalm#3886]
Psalm reports MissingFile error when using `include` or `require` on a file that does not exist. This is normal, because you shouldn't be including unexisting file.

However, it also reports the same error, even if you used `is_file` to check if file exists and skip `include`/`require` if it doesn't. I think that this shouldn't cause error, because it also won't cause error when actually running script in PHP. Checking if file exists is useful if you want to include some user settings, which are not guaranteed and required to exist, and most likely won't exist when you are running Psalm in CI.

Example snippets which fail:

```php
<?php

$settings = is_file(__DIR__ . '/settings.php') ? require __DIR__ . '/settings.php' : [];
```

```php
<?php

if (is_file(__DIR__ . '/settings.php')) {
    $settings = require __DIR__ . '/settings.php';
} else {
    $settings = [];
}
```

This causes:

```
ERROR: MissingFile - test1.php:3:50 - Cannot find file /tmp/psalm-test/settings.php to include (see https://psalm.dev/107)
$settings = is_file(__DIR__ . '/settings.php') ? require __DIR__ . '/settings.php' : [];


ERROR: MissingFile - test2.php:4:17 - Cannot find file /tmp/psalm-test/settings.php to include (see https://psalm.dev/107)
    $settings = require __DIR__ . '/settings.php';
```

I can't show example on https://psalm.dev because file includes are not allowed there.

**Comment by muglug** [comment]
The issue here is that Psalm can't figure out what you're trying to include (because it wants to analyse the contents). Feel free to ignore the issue.
